packages/remix-dev/cli/migrate/types.ts

```diff
@@ -1,9 +1,11 @@
 import type { Flags } from "./flags";
 
-export type MigrationFunction = (args: {
-  flags: Flags;
-  projectDir: string;
-}) => Promise<void>;
+export type MigrationFunction = (
+  projectDir: string,
+  options?: {
+    flags: Flags;
+  }
```

Comment:
nit: `flags` are already a map of options, so flattening this would simplify. If we need non-flag options later, then we can add the `options` wrapper, but right now we don't need it:

```suggestion
  flags?: Flags,
```